### PR TITLE
Fix credential issue in hlc

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1724

- Add default credential info if the `addCredentials` is set as true by customers
- Add UT for readme generation in HLC

### CODE REVIEW

File: packages/autorest.typescript/src/generators/clientFileGenerator.ts [hunk 1]
```diff
@@ -647,11 +654,8 @@ function getTrack2DefaultContent(
   packageDetails: PackageDetails,
   clientDetails: ClientDetails
 ) {
-  const {
-    azureArm,
-    allowInsecureConnection,
-    addCredentials
-  } = getAutorestOptions();
+  const { azureArm, allowInsecureConnection } = getAutorestOptions();
+  const { addCredentials } = getSecurityInfoFromModel(clientDetails.security);
```
Comment (MaryGao) [inline packages/autorest.typescript/src/generators/clientFileGenerator.ts:658]: Here is the only change for this pr, others are formattings. We get the option `addCredentials` from `getSecurityInfoFromModel` not `getAutorestOptions`.

Comment (qiaozha) [reply packages/autorest.typescript/src/generators/clientFileGenerator.ts:658]: I remember there's a case where we set add-credentials true for future manual customization and don't have a security schema defined.

Comment (MaryGao) [reply packages/autorest.typescript/src/generators/clientFileGenerator.ts:658]: Currently fix we would ignore the credential info if there is security schema defined. If we'd like to support this case we need to decide how to generate the code do you have more info for that case?

File: packages/autorest.typescript/src/utils/schemaHelpers.ts [hunk 2]
```diff
@@ -258,13 +258,21 @@ export function getSecurityInfoFromModel(security: Security) {
...
   }
-  const scopes: string[] = [];
-  credentialScopes.forEach(item => {
-    scopes.push(item);
-  });
+  const scopes: string[] = [...credentialScopes];
+  let refinedAddCredentials =
+    addCredentials === false ? false : security.authenticationRequired;
+  // Add default info if explictly set the addCredentials as true
+  if (
```
Comment (MaryGao) [inline packages/autorest.typescript/src/utils/schemaHelpers.ts:265]: Set the default value as token credential with scope `https://management.azure.com/.default`.

**Review by qiaozha — APPROVED** [review]

**Pull request merged**